server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

```diff
@@ -541,27 +541,43 @@ private <T extends AbstractIdentityProviderDefinition> IdentityProvider<?> ensur
             return originalIdp;
         }
 
+        final IdentityProvider<?> persistedAliasIdp = createNewAliasIdp(aliasIdp, originalIdp.getAliasZid());
```

Comment:
I like the change reducing the length of a long method. I'm curious what the Sonar smells are that all of these changes are addressing.

Reply:
This is the fix to the Sonar warning about the "Cognitive Complexity" being too high in this method.